pkg/runutil/runutil.go

```diff
@@ -164,14 +164,17 @@ func ExhaustCloseWithErrCapture(err *error, r io.ReadCloser, format string, a ..
 // dir except for the ignoreDirs directories.
 func DeleteAll(dir string, ignoreDirs ...string) error {
 	entries, err := ioutil.ReadDir(dir)
+	if os.IsNotExist(err) {
```

Comment:
That is an important change. We are making `DeleteAll` idempotent which has some perks but also allows us to spam `DeleteAll` function where we does not need it 🤗 that's why we did not have it first place. Can we avoid this?

Reply:
Actually there is bigger discussion you can join (: 

https://github.com/thanos-io/thanos/issues/3662

Reply:
It's related to bucket API delete but same problem

Reply:
We can check in advance whether the directory exists or not. Potentially we could only check the first compaction group but that's not a given because `--compaction.concurrency` could be more than `1`. Do you think this would be a better approach?

Reply:
Proposed something.